packages/graphql/src/schema/validation/validate-document.ts

```diff
@@ -180,15 +181,24 @@ function validateDocument(
         ],
     });
 
-    const schema = extendSchema(schemaToExtend, doc);
-
-    const errors = validateSchema(schema);
-
-    const filteredErrors = errors.filter((e) => e.message !== "Query root type must be provided.");
+    return extendSchema(schemaToExtend, doc, { assumeValid: !validateTypeDefs });
+}
 
-    if (filteredErrors.length) {
-        throw new Error(filteredErrors.join("\n"));
+function validateDocument(
+    document: DocumentNode,
+    validateTypeDefs = true,
+    additionalDirectives: Array<GraphQLDirective> = [],
+    additionalTypes: Array<GraphQLNamedType> = []
+): GraphQLSchema {
```

Comment:
With the suggestion below, this will be able to remain as a non-returning function which throws if necessary:

```suggestion
): void {
```